docs/cluster-agent/HORIZONTAL_POD_AUTOSCALING.md

```diff
@@ -0,0 +1,262 @@
+To this extent, should you want to further proceed, we are assuming that:
+1. You have node agents running (ideally from a DaemonSet) with the Autodiscovery process enabled and functional.
+2. Agents are configured to securely (see [this section](/Dockerfiles/cluster-agent/README.md#security-premise) of the official documentation) communicate with the DCA.
+
+The Second point is not mandatory, but it enables the enrichment of the metadata collected by the node agents. 
+
+### Spinning up the DCA
+
+In order to spin up the DCA, you will need to create the appropriate RBAC rules.
+The DCA is acting as a proxy between the API Server and the node agent, to this extent it will need to have access to some cluster level resources.
+
+`kubectl apply -f manifests/cluster-agent/rbac-cluster-agent.yaml`
+
+```
+clusterrole.rbac.authorization.k8s.io "dca" created
+clusterrolebinding.rbac.authorization.k8s.io "dca" created
+serviceaccount "dca" created
+```
+
+Then you will need to create the DCA and its services.
+Start by adding your API_KEY and APP_KEY in the deployment manifest of the DCA.
+Then enable the HPA Processing by setting the `DD_ENABLE_HPA` variable to true.
+Finally, spin up the resources:
+
+`kubectl apply -f manifests/cluster-agent/cluster-agent.yaml`
+`kubectl apply -f manifests/cluster-agent/datadog-cluster-agent_service.yaml`
+`kubectl apply -f manifests/cluster-agent/hpa-example/cluster-agent-hpa-svc.yaml`
+
+Note that the first service is used for the communication between the node agents and the DCA but the second is used by Kubernetes to register the External Metrics Provider.
+
+At this point you should be having:
+
+```
+PODS:
+
+NAMESPACE     NAME                                     READY     STATUS    RESTARTS   AGE
+default       datadog-cluster-agent-7b7f6d5547-cmdtc   1/1       Running   0          28m
+
+SVCS:
+
+NAMESPACE     NAME                  TYPE        CLUSTER-IP        EXTERNAL-IP   PORT(S)         AGE
+default       datadog-custom-metrics-server   ClusterIP   192.168.254.87    <none>        443/TCP         28m
+default       dca                   ClusterIP   192.168.254.197   <none>        5005/TCP        28m
+
+```
+
+### Register the External Metrics Provider
+
+Once the DCA is up and running, you can register it as an External Metrics Provider, via the service exposing the port 443.
+
+To do so, simply apply the following RBAC rules:
+
+`kubectl apply -f manifest/hpa-example/rbac-hpa.yaml`
+
+```
+clusterrolebinding.rbac.authorization.k8s.io "system:auth-delegator" created
+rolebinding.rbac.authorization.k8s.io "dca" created
+apiservice.apiregistration.k8s.io "v1beta1.external.metrics.k8s.io" created
+clusterrole.rbac.authorization.k8s.io "external-metrics-reader" created
+clusterrolebinding.rbac.authorization.k8s.io "external-metrics-reader" created
+```
+ 
+Once you have the DCA running and the service registered, you can create an HPA manifest and let the DCA pull metrics from Datadog.
```

Comment:
r/`the service registered, you can create an `/`the service registered, create an`